### PR TITLE
Improve getopts and stderr message consistency in main

### PR DESCRIPTION
**Pull request by roderickvd**
There is no definitive standard on how to format getopts strings.
Usage of curly brackets seems best practice by these standards:
- http://www.tfug.org/helpdesk/general/man.html
- https://developers.google.com/style/code-syntax

This is cosmetic only and was mostly extracted from #685.

### CODE REVIEW

**Comment by JasonLG1979** [comment]
Not really an "options" issue but more of an unexpected missing arg. There is no `--help` to list all possible args. Granted if you try `librespot --help` it will complain with `error: Unrecognized option: 'help'` and then proceed to list all possible args. It would be nice if librespot had a proper help (`-h`, `--help`) It could be a separate PR but it seems like an oversight to not have a help.

**Comment by roderickvd** [comment]
Here you go.

File: src/main.rs [hunk 1]
```diff
@@ -191,175 +191,188 @@ struct Setup {
 
 fn get_setup(args: &[String]) -> Setup {
     let mut opts = getopts::Options::new();
-    opts.optopt(
+    opts.optflag(
+        "h",
+        "help",
+        "Print this help menu",
```
Comment (Johannesd3) [inline src/main.rs:197]: Maybe
```suggestion
        "Print this help menu.",
```

File: src/main.rs [hunk 2]
```diff
@@ -191,175 +191,188 @@ struct Setup {
...
+    )
+    .optflag(
+        "",
+        "mixer-linear-volume",
+        "Disable Alsa mixer mapped volume scale (cubic). Defaults to false.",
+    )
+    .optopt(
+        "",
+        "initial-volume",
+        "Initial volume (%) once connected {0..100}. Defaults to 50.",
```
Comment (Johannesd3) [inline src/main.rs:281]: The new one is slightly more confusing.

File: src/main.rs [hunk 3]
```diff
@@ -191,175 +191,188 @@ struct Setup {
...
+        "Pass raw stream to output, only works for pipe and subprocess.",
+    );
 
     let matches = match opts.parse(&args[1..]) {
         Ok(m) => m,
         Err(f) => {
-            eprintln!("error: {}\n{}", f.to_string(), usage(&args[0], &opts));
+            eprintln!(
+                "Error parsing command line options: {}\n{}",
+                f.to_string(),
```
Comment (Johannesd3) [inline src/main.rs:364]: Why `to_string`?

File: src/main.rs (hunk 1 above)
Comment (roderickvd) [reply src/main.rs]: Fixed in 2c9ac40 🤦

File: src/main.rs (hunk 2 above)
Comment (roderickvd) [reply src/main.rs]: Changed to `Initial volume once connected {0-100}%. Defaults to 50 for softvol and for Alsa mixer the current volume.` in 2c9ac40. Better that way?

File: src/main.rs (hunk 3 above)
Comment (roderickvd) [reply src/main.rs]: Removed in c405585

File: src/main.rs (hunk 2 above)
Comment (Johannesd3) [reply src/main.rs]: > Initial volume

seems fine

> once connected

what does that mean?

> {0-100}%.

why curly braces. And do I have to add the "%" sign in my input.

> Defaults to 50 for softvol and for Alsa mixer the current volume.

Nice chiasmus. But I'd prefer a more regular syntax.

Comment (roderickvd) [reply src/main.rs]: No worries I appreciate the vetting. I had to look up chiasmus. 😄 How about:
`Initial volume in % from 0-100. Default for softvol: '50'. For the Alsa mixer: the current volume.`

Comment (Johannesd3) [reply src/main.rs]: Sounds good

Comment (roderickvd) [reply src/main.rs]: Done in 3e2a71e

**Pull request merged**